controllers/build_controller.go

```diff
@@ -0,0 +1,317 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+// BuildReconciler reconciles a Build Pod object
+type BuildReconciler struct {
+	client.Client
+	Scheme            *runtime.Scheme
+	InsightsJWTSecret string
+	ScanImageName     string
+}
+
+const insightsBuildPodScannedLabel = "insights.lagoon.sh/scanned"
+const insightsScanPodLabel = "insights.lagoon.sh/scan-status"
+const dockerhost = "docker-host.lagoon.svc" //TODO in future versions this will be read from the build CRD
+
+//+kubebuilder:rbac:groups=core,resources=deployments,verbs=get;list
+//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=namespaces,verbs=get;list
+//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list
```

Comment:
might be worth running `make manifests` to redo all the templates (it will help when it comes time to updating the helm charts)